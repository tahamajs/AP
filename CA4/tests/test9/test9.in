report_salaries
report_employee_salary 5
report_employee_per_hour 7 12
report_team_salary 2
report_team_salary 1
update_team_bonus 1 25
report_team_salary 1
report_total_hours_per_day 5 16
update_team_bonus 4 15
report_salaries
report_employee_salary 7
report_team_salary 3
add_working_hours 8 14 0 13
delete_working_hours 2 1
report_employee_per_hour 2 33
show_salary_config team_lead
add_working_hours 4 26 5 12
update_team_bonus 3 17
add_working_hours 8 17 7 23
report_team_salary 1
report_employee_salary 4
report_total_hours_per_day 10 15
report_employee_salary 6
update_salary_config senior 2200 35 - - 26
report_total_hours_per_day 7 1
report_team_salary 3
update_salary_config manager - 40 - 200 -
update_team_bonus 2 30
report_salaries
delete_working_hours 5 10
delete_working_hours 7 0
report_employee_per_hour 19 13
add_working_hours 7 3 18 17
report_team_salary 3
report_salaries
show_salary_config junior
report_employee_per_hour 1 23
report_salaries
