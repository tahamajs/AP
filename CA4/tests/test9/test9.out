ID: 0
Name: Fateme Shakeri
Total Working Hours: 200
Total Earning: 9520
---
ID: 3
Name: ZahraHasani
Total Working Hours: 228
Total Earning: 13114
---
ID: 4
Name: Fariba
Total Working Hours: 202
Total Earning: 12977
---
ID: 5
Name: maysaAsgari
Total Working Hours: 226
Total Earning: 16360
---
ID: 7
Name: Iman Karimi
Total Working Hours: 217
Total Earning: 15942
---
ID: 8
Name: BardiaPiltan
Total Working Hours: 175
Total Earning: 10875
---
ID: 5
Name: maysaAsgari
Age: 39
Level: team_lead
Team ID: 2
Total Working Hours: 226
Absent Days: 0
Salary: 23372
Bonus: 0
Tax: 7012
Total Earning: 16360
7-8: 1.9
8-9: 2.3
9-10: 2.3
10-11: 2.6
11-12: 2.5
---
Period(s) with Max Working Employees: 10-11
Period(s) with Min Working Employees: 7-8
ID: 2
Head ID: 5
Head Name: maysaAsgari
Team Total Working Hours: 454
Average Member Working Hours: 227.0
Bonus: 0
---
Member ID: 3
Total Earning: 13114
---
Member ID: 5
Total Earning: 16360
---
ID: 1
Head ID: 7
Head Name: Iman Karimi
Team Total Working Hours: 592
Average Member Working Hours: 197.3
Bonus: 0
---
Member ID: 0
Total Earning: 9520
---
Member ID: 7
Total Earning: 15942
---
Member ID: 8
Total Earning: 10875
---
OK
ID: 1
Head ID: 7
Head Name: Iman Karimi
Team Total Working Hours: 592
Average Member Working Hours: 197.3
Bonus: 25
---
Member ID: 0
Total Earning: 11900
---
Member ID: 7
Total Earning: 19928
---
Member ID: 8
Total Earning: 13594
---
Day #5: 54
Day #6: 40
Day #7: 52
Day #8: 34
Day #9: 39
Day #10: 46
Day #11: 47
Day #12: 31
Day #13: 48
Day #14: 32
Day #15: 53
Day #16: 48
---
Day(s) with Max Working Hours: 5
Day(s) with Min Working Hours: 12
TEAM_NOT_FOUND
ID: 0
Name: Fateme Shakeri
Total Working Hours: 200
Total Earning: 11900
---
ID: 3
Name: ZahraHasani
Total Working Hours: 228
Total Earning: 13114
---
ID: 4
Name: Fariba
Total Working Hours: 202
Total Earning: 12977
---
ID: 5
Name: maysaAsgari
Total Working Hours: 226
Total Earning: 16360
---
ID: 7
Name: Iman Karimi
Total Working Hours: 217
Total Earning: 19928
---
ID: 8
Name: BardiaPiltan
Total Working Hours: 175
Total Earning: 13594
---
ID: 7
Name: Iman Karimi
Age: 26
Level: team_lead
Team ID: 1
Total Working Hours: 217
Absent Days: 2
Salary: 22775
Bonus: 5694
Tax: 8541
Total Earning: 19928
TEAM_NOT_FOUND
OK
EMPLOYEE_NOT_FOUND
INVALID_ARGUMENTS
Base Salary: 6500
Salary Per Hour: 75
Salary Per Extra Hour: 62
Official Working Hours: 220
Tax: 30%
INVALID_INTERVAL
TEAM_NOT_FOUND
OK
ID: 1
Head ID: 7
Head Name: Iman Karimi
Team Total Working Hours: 621
Average Member Working Hours: 207.0
Bonus: 25
---
Member ID: 0
Total Earning: 11900
---
Member ID: 7
Total Earning: 19928
---
Member ID: 8
Total Earning: 15225
---
ID: 4
Name: Fariba
Age: 32
Level: expert
Team ID: N/A
Total Working Hours: 202
Absent Days: 5
Salary: 17536
Bonus: 0
Tax: 4559
Total Earning: 12977
Day #10: 46
Day #11: 47
Day #12: 31
Day #13: 48
Day #14: 45
Day #15: 53
---
Day(s) with Max Working Hours: 15
Day(s) with Min Working Hours: 12
EMPLOYEE_NOT_FOUND
OK
INVALID_ARGUMENTS
TEAM_NOT_FOUND
INVALID_LEVEL
OK
ID: 0
Name: Fateme Shakeri
Total Working Hours: 200
Total Earning: 11900
---
ID: 3
Name: ZahraHasani
Total Working Hours: 228
Total Earning: 9918
---
ID: 4
Name: Fariba
Total Working Hours: 202
Total Earning: 12977
---
ID: 5
Name: maysaAsgari
Total Working Hours: 226
Total Earning: 21269
---
ID: 7
Name: Iman Karimi
Total Working Hours: 217
Total Earning: 19928
---
ID: 8
Name: BardiaPiltan
Total Working Hours: 204
Total Earning: 8639
---
OK
INVALID_ARGUMENTS
INVALID_ARGUMENTS
INVALID_ARGUMENTS
TEAM_NOT_FOUND
ID: 0
Name: Fateme Shakeri
Total Working Hours: 200
Total Earning: 11900
---
ID: 3
Name: ZahraHasani
Total Working Hours: 228
Total Earning: 9918
---
ID: 4
Name: Fariba
Total Working Hours: 202
Total Earning: 12977
---
ID: 5
Name: maysaAsgari
Total Working Hours: 217
Total Earning: 20726
---
ID: 7
Name: Iman Karimi
Total Working Hours: 217
Total Earning: 19928
---
ID: 8
Name: BardiaPiltan
Total Working Hours: 204
Total Earning: 8639
---
Base Salary: 1900
Salary Per Hour: 50
Salary Per Extra Hour: 30
Official Working Hours: 210
Tax: 20%
1-2: 0.6
2-3: 1.0
3-4: 1.2
4-5: 1.3
5-6: 1.5
6-7: 1.7
7-8: 2.0
8-9: 2.3
9-10: 2.4
10-11: 2.6
11-12: 2.6
12-13: 2.7
13-14: 2.7
14-15: 2.8
15-16: 2.8
16-17: 2.5
17-18: 2.3
18-19: 1.8
19-20: 1.6
20-21: 1.5
21-22: 1.1
22-23: 0.7
---
Period(s) with Max Working Employees: 14-15 15-16
Period(s) with Min Working Employees: 1-2
ID: 0
Name: Fateme Shakeri
Total Working Hours: 200
Total Earning: 11900
---
ID: 3
Name: ZahraHasani
Total Working Hours: 228
Total Earning: 9918
---
ID: 4
Name: Fariba
Total Working Hours: 202
Total Earning: 12977
---
ID: 5
Name: maysaAsgari
Total Working Hours: 217
Total Earning: 20726
---
ID: 7
Name: Iman Karimi
Total Working Hours: 217
Total Earning: 19928
---
ID: 8
Name: BardiaPiltan
Total Working Hours: 204
Total Earning: 8639
---
